[EN-1388] Moves python27 role to own repo

agirivera opened the pull request:
Story: http://jira.juwai.com/browse/EN-1388
Sub-task: http://jira.juwai.com/browse/EN-1403
- Adds `amazon_python_packages` to configuration
- Adds `juwai.common` as a dependency
- Fixes `executable-pip` from `pip-2.7` to `pip2.7`

To test:
1. Clone this repo `git clone https://github.com/agirivera/ansible-role-python27.git python27`
2. Go to test directory `cd python27/tests`
3. Install dependencies `ansible-galaxy install -p roles -r requirements.yml`
4. Provision a test box `vagrant up`

/cc @juwai/platform @imlazyone @tjoelsson Please review. Thanks.

Commit:
[EN-1388] Moves python27 role to own repo

tjoelsson commented on tests/Vagrantfile:
```diff
@@ -0,0 +1,14 @@
+
+Vagrant.configure(2) do |config|
+  config.vm.box = "centos65"
+  config.vm.box_url = "http://downloads.juwai.io/devops/boxes/centos65-x86-64.box"
```
Is this URL public?

agirivera replied:
Hmm. You're right, we probably don't want that known outside?

agirivera replied:
Do we have anything private on that box? It seems to be a pretty basic install of centos65.

stuartwan replied:
It is just a simple CentOS65 box. That box url is now public I think.

tjoelsson replied:
How about just use this one https://atlas.hashicorp.com/nrel/boxes/CentOS-6.5-x86_64/versions/1.2.0 ?
Or this https://developer.nrel.gov/downloads/vagrant-boxes/CentOS-6.5-x86_64-v20140504.box ?

stuartwan replied:
Why do we need to change another box? the current one is too big?

tjoelsson replied:
I don't know why we have our own box in the first place. Do we want people to use it?

stuartwan replied:
At first we think with our own box we can put some commonly used stuff on it, and we can have a smaller size...  (for now I'm not sure if that's still the case.)

Commit:
[EN-1388] Removes accidental commit

tjoelsson commented:
Please tell us if you changed anything when moving, as we can not see a diff here.

agirivera commented:
@tjoelsson: Sorry, will do next time. I've removed an error output file which accidentally got committed.

tjoelsson commented:
@agirivera then how to review this PR?

agirivera commented:
@tjoelsson: Not sure what you meant, but I've got no more changes. Please go ahead with review.

tjoelsson commented:
@agirivera I mean, you copied this role from ansible-playbooks repo. I can't see if you changed anything after copying. If you changed some line of code, can you please tell me where?

tjoelsson commented on README.md:
```diff
@@ -1,2 +1,59 @@
+Dependencies
+------------
+
+None.
```
It depends on juwai.common

agirivera replied:
Good catch :smile_cat: Updated.

tjoelsson commented:
? I like new lines at end of files

Commit:
[EN-1388] Adds blank lines at the end of files and updates README.md
against the dependency requirements

agirivera commented:
> ? I like new lines at end of files

Happy to oblige :smile_cat: Done.

tjoelsson commented:
The test instructions only work if you name your cloned directory exactly `python27`. I usually leave mine the same as the repo. To fix this, I suggest:
1. Remove the roles symlink
2. Create a real roles directory
3. Create python27 symlink in the roles directory, pointing to the root directory

Commit:
[EN-1388] Restructures roles/ for testing

agirivera commented:
@tjoelsson: Adjusted `roles` as suggested :smile_cat:

tjoelsson commented:
When I run the test according to instructions here is the output from provisioning:

```
PYTHONUNBUFFERED=1 ANSIBLE_FORCE_COLOR=true ANSIBLE_HOST_KEY_CHECKING=false ANSIBLE_SSH_ARGS='-o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ControlMaster=auto -o ControlPersist=60s' ansible-playbook --connection=ssh --timeout=30 --limit='default' --inventory-file=/Users/Tomas/ansible-role-python27/tests/.vagrant/provisioners/ansible/inventory -v test.yml
No config file found; using defaults

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [default]

PLAY RECAP *********************************************************************
default                    : ok=1    changed=0    unreachable=0    failed=0   
```

Looks like it didn't do anything.

Commit:
[EN-1388] Fixes role link target

agirivera commented:
@tjoelsson: I made a mistake with the link, incorrectly added an extra `../` on the link target which caused nothing to happen. Already fixed. Thanks.

tjoelsson commented:
It complains that I don't have the `juwai.common` role. How about adding a requirements.yml in the tests folder?

Commit:
[EN-1388] Adds requirements file for successful local test

Commit:
[EN-1388] Adds blank line at the end

agirivera commented:
@tjoelsson Added `tests/requirements.yml` and updated test steps in description above.

stuartwan commented:
? Do we really need a `Vagrantfile` in a Role?

tjoelsson commented:
Instead of using sudo to install roles globally, I suggest the following:

```
$ ansible-galaxy install -p roles -r requirements.yml
```

stuartwan commented:
Or, shall we put a Vagrantfile in every role under tests/ ?

tjoelsson commented:
@imlazyone do you have another suggestion?

stuartwan commented:
> @imlazyone do you have another suggestion?

Regarding role installation or Vagrantfile?

tjoelsson commented:
@imlazyone Vagrantfile

stuartwan commented:
I can see the benefit on testing with the Vagrantfile, but, all of our other roles don't have a vagrantfile... 

Just thinking about consistency.

tjoelsson commented:
When this is deployed, I want to add the same test file structure to the two public roles I worked on.

agirivera commented:
@imlazyone: We probably won't need the vagrant file eventually, but the idea is to be able to test our role locally before pushing or travis kicks in. Right?

As for consistency, I don't think it should be an issue now as long as we have ways to test the other roles. Mine's just dependent on the box and it seems to be a good idea to add this now rather than having you guys create the file again locally.

Commit:
[EN-1388] Adds control file to ignore locally installed roles for
testing

agirivera commented:
@tjoelsson Steps for testing has been updated to install dependency roles locally, and added a control file to ignore roles installed this way from revisioning.

tjoelsson commented:
👍

stuartwan commented on README.md:
```diff
@@ -1,2 +1,59 @@
+License
+-------
+
+MIT / BSD
```
? I suggest we use only MIT for convenience and consistency. FYI you can add a `LICENSE` file in the root folder just like [here](https://github.com/juwai/ansible-role-supervisor).

agirivera replied:
? How about a separate story to go through the roles and _consistencize_ the licenses? We should be only moving the roles for now.

agirivera replied:
Updated licensing info. Thanks :smile_cat:

imlazyone commented on defaults/main.yml:
```diff
@@ -0,0 +1,16 @@
+---
+py_version: 2.7.9 # same as AWS
```
? I'm not sure if 2.7.9 is the current AWS latest. I logged onto one of our RPC production servers, the Python version in juwai-rpc virtualenv is 2.7.10. (And latest Python 2.7 minor version is 2.7.11).

agirivera replied:
2.7.9 was the latest I think when the role was first written :smile_cat: This variable is only used when we are compiling our own on the vagrant boxes I think. We're only moving the roles into their own repos so I believe we're safe for the moment and we can create a story to update this and test for breakages.

Commit:
[EN-1388]

- Adds yaml header to meta/main.yml
- Adds placeholder files in role folders to include them for tracking

Commit:
[EN-1388] Makes MIT the sole license

tjoelsson commented:
👍

stuartwan commented:
:+1:

agirivera commented:
Awesome :smile_cat: Please merge. Who has access to the galaxy account by the way?
